src/partition/repart.c

```diff
@@ -1407,8 +1415,16 @@ static int context_read_definitions(
                 return log_error_errno(r, "Failed to enumerate *.conf files: %m");
 
         STRV_FOREACH(f, files) {
+                char **conf_file_dirs;
+                _cleanup_strv_free_ char **drop_in_files = NULL;
                 _cleanup_(partition_freep) Partition *p = NULL;
 
+                if (directory)
+                        conf_file_dirs = STRV_MAKE(directory);
+                else {
+                        conf_file_dirs = CONF_PATHS_STRV("repart.d");
+                }
```

Comment:
We should reduce duplication by only calculating the list of directories to use once, since it's already done for the first call to `conf_files_list_strv()` earlier in this function as well. e.g. I'd do something like the following at the start of this function:

```
const char *const *dirs;

if (directory)
        dirs = STRV_MAKE_CONST(directory);
else
        dirs = (const char* const*) CONF_PATHS_STRV("repart.d");
```

Also remember that single line if statements should not have accompanying braces.